src/main/java/org/reaktivity/nukleus/http_cache/internal/stream/HttpCacheProxyNonCacheableResponse.java

```diff
@@ -0,0 +1,262 @@
+    private int padding;
+
+    private long traceId;
+
+    HttpCacheProxyNonCacheableResponse(
+        HttpCacheProxyFactory httpCacheProxyFactory,
+        HttpCacheProxyRequest request,
+        MessageConsumer connectReplyThrottle,
+        long connectRouteId,
+        long connectReplyId,
+        long acceptInitialId)
+    {
+        this.streamFactory = httpCacheProxyFactory;
+        this.request = request;
+        this.connectReplyThrottle = connectReplyThrottle;
+        this.connectRouteId = connectRouteId;
+        this.connectReplyStreamId = connectReplyId;
+        this.acceptInitialId = acceptInitialId;
+        this.streamState = this::onResponseMessage;
+    }
+
+    public void onResponseMessage(
+        int msgTypeId,
+        DirectBuffer buffer,
+        int index,
+        int length)
+    {
+        switch (msgTypeId)
+        {
+            case BeginFW.TYPE_ID:
+                final BeginFW begin = this.streamFactory.beginRO.wrap(buffer, index, index + length);
+                handleBegin(begin);
+                break;
+            case WindowFW.TYPE_ID:
+                final WindowFW window = streamFactory.windowRO.wrap(buffer, index, index + length);
+                onWindowWhenProxying(window);
+                break;
+            case ResetFW.TYPE_ID:
+                final ResetFW reset = streamFactory.resetRO.wrap(buffer, index, index + length);
+                onResetWhenProxying(reset);
+                break;
+            default:
+                // ignore
+                break;
+        }
+    }
+
+    void handleStream(
+        int msgTypeId,
+        DirectBuffer buffer,
+        int index,
+        int length)
+    {
+        streamState.accept(msgTypeId, buffer, index, length);
+    }
+
+    @Override
+    public String toString()
+    {
+        return String.format("%s[connectRouteId=%016x, connectReplyStreamId=%d, acceptReplyBudget=%016x, " +
+                " padding=%d]", getClass().getSimpleName(),
+            connectRouteId, connectReplyStreamId, acceptReplyBudget, padding);
+    }
+
+
+    private void handleBegin(
+        BeginFW begin)
+    {
+        final long connectReplyId = begin.streamId();
+        traceId = begin.trace();
+
+        final OctetsFW extension = streamFactory.beginRO.extension();
+
+        if (extension.sizeof() > 0)
+        {
+            final HttpBeginExFW httpBeginFW = extension.get(streamFactory.httpBeginExRO::wrap);
+
+            if (DEBUG)
+            {
+                System.out.printf("[%016x] CONNECT %016x %s [received response]\n", currentTimeMillis(), connectReplyId,
+                        getHeader(httpBeginFW.headers(), ":status"));
+            }
+
+            final ListFW<HttpHeaderFW> responseHeaders = httpBeginFW.headers();
+            doProxyBegin(responseHeaders);
+        }
+        else
+        {
+            this.streamFactory.writer.doReset(connectReplyThrottle, connectRouteId, connectReplyStreamId, traceId);
+        }
```

Comment:
Can we test this at stream creation time instead, so that `RESET` is handled implicitly by `reaktor` when returned stream is `null`?

Reply:
I removed everywhere, I made a not for myself to test it.